src/main/java/com/scottlogic/filters/DatePostFilter.java

```diff
@@ -0,0 +1,30 @@
+package com.scottlogic.filters;
+
+import com.scottlogic.PostFilter;
+import com.scottlogic.UserPost;
+
+import java.time.OffsetDateTime;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Objects;
+
+public class DatePostFilter implements PostFilter {
+    private final OffsetDateTime startDate;
+    private final OffsetDateTime endDate;
+
+    public DatePostFilter(OffsetDateTime startDateInput, OffsetDateTime endDateInput) {
+        startDate = startDateInput;
+        endDate = endDateInput;
+    }
+
+    @Override
+    public List<UserPost> filter(List<UserPost> inputList) {
+        if (inputList == null) {
+            return null;
+        }
+        List<UserPost> outputList = new ArrayList<>(inputList);
+        outputList.removeIf(a -> !(a.getDateTime().compareTo(startDate) > 0));
```

Comment:
You have used quite a few `!` with `<` which you could always refactor to `>=`  along with `>` to `<=`, this might just help save a few extra brackets and in general make the code look cleaner